File: lib/core/models/bank.dart
class Bank {
  final String id;
  final String name;
  final String slogan;
  final double minimumBalance;
  final double withdrawFees;
  final double internalTransferFees;
  final double externalTransferFees;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Bank({
    required this.id,
    required this.name,
    required this.slogan,
    required this.minimumBalance,
    required this.withdrawFees,
    required this.internalTransferFees,
    required this.externalTransferFees,
    required this.createdAt,
    this.updatedAt,
  });

  factory Bank.fromJson(Map<String, dynamic> json) {
    return Bank(
      id: json['id'],
      name: json['name'],
      slogan: json['slogan'],
      minimumBalance: (json['minimumBalance'] as num).toDouble(),
      withdrawFees: (json['withdrawFees'] as num).toDouble(),
      internalTransferFees: (json['internalTransferFees'] as num).toDouble(),
      externalTransferFees: (json['externalTransferFees'] as num).toDouble(),
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: json['updatedAt'] != null ? DateTime.tryParse(json['updatedAt']) : null,
    );
  }
}

File: service-banks/src/main/java/com/jamaa/banks/entity/Bank.java
package com.jamaa.banks.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "banks")
@NoArgsConstructor
@Getter
@Setter
public class Bank {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column
    private String slogan;

    @Column(name = "logo_url")
    private String logoUrl;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "minimum_balance", nullable = false)
    private BigDecimal minimumBalance;

    @Column(name = "withdraw_fees", nullable = false)
    private BigDecimal withdrawFees;

    @Column(name = "internal_transfer_fees", nullable = false)
    private BigDecimal internalTransferFees;

    @Column(name = "external_transfer_fees", nullable = false)
    private BigDecimal externalTransferFees;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
} 